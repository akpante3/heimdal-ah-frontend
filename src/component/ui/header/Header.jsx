import React from 'react';
import PropTypes from 'prop-types';
import './Header.scss';
import NavItems from '../navItems/NavItems';
import ActiveUser from '../Widgets/activeuser/ActiveUser';
import Logo from '../logo/Logo';
import SearchForm from '../Widgets/searchForm/SearchForm';
import HeaderButton from '../Widgets/headerButton/HeaderButton';

/**
 * @param {boolean} isValidated should be a boolean
 */
const Header = (props) => {
  const { isValidated } = props;
  return (
    <header className='header'>
      <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <Logo />
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav mr-auto">
            <NavItems />
          </ul>
          {isValidated ? <ActiveUser /> : <HeaderButton />}
          <SearchForm />
        </div>
      </nav>
    </header>
  );
};

Header.propTypes = {
  isValidated: PropTypes.bool.isRequired,
};

export default Header;
