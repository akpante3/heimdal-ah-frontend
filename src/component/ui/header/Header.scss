$cardo: "Cardo", serif;
$dropdown-content: #f1f1f1;
$dropdown-content-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
$primary-color: #606e86;
$secondary-color: #ffff;
$reem-kufi: "Reem Kufi", sans-serif;

.header {
  font-family: $reem-kufi;

  .navbar {
    background: $secondary-color;
    box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.1);
  }

  .bg-light {
    background-color: $secondary-color !important;
  }

  .header-form {
    padding: 10px;
    margin-right: 20px;
    margin-top: -10px !important;
  }

  .header-input {
    display: none !important;
  }

  .active_user_logo {
    margin-right: 70px;
  }

  .notification_icon {
    margin-right: 60px;
    width: 35px;
  }

  .profile-icon {
    margin-left: 70px;
    margin-top: 20px;
    margin-bottom: 20px;
    width: 30px;
  }

  .search-input {
    margin-right: 10px !important;
    width: 150px;
  }

  .search_icon {
    -webkit-transform: scaleX(-1);
    color: $primary-color;
    cursor: pointer;
    transform: scaleX(-1);
  }

  .username {
    font-size: 18px;
    padding: 10px;

    &::after {
      border-bottom: 1px solid $primary-color;
      content: "";
      display: block;
      margin: 0 auto;
      margin-top: 12px;
      width: 90%;
    }
    text-align: center;
  }

  .dropdown {
    display: inline-block;
    position: relative;
  }

  .dropdown-content {
    background-color: $dropdown-content;
    box-shadow: $dropdown-content-shadow;
    display: none;
    font-family: "Cardo", serif;
    min-width: 160px;
    position: absolute;
    z-index: 1;

    a {
      color: $primary-color;
      display: block;
      padding: 12px 16px;
      text-decoration: none;
      &:hover {
        background-color: $primary-color;
        color: $secondary-color;
      }
    }
  }

  .user-settings::before {
    border-top: 1px solid $primary-color;
    content: "";
    display: block;
    margin: 0 auto;
    width: 90%;
  }

  .dropdown:hover .dropdown-content {
    display: block;
    margin-right: 60px;
  }
}
