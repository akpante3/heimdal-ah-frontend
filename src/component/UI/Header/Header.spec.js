/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import toJson from 'enzyme-to-json';
import Header from './Header';

describe('Testing the App', () => {
  it('should render the Routes', () => {
    const wrapper = shallow(<Header />);
    const nav = wrapper.find('nav');
    expect(nav.length).toBe(1);
  });

  it('should match the snapshot', () => {
    const tree = shallow(<Header />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  // it('simulates click events', () => {
  //   const onButtonClick = sinon.spy();
  //   const wrapper = shallow(<Header />);
  //   const searchIcon = wrapper.find('');

  //   searchIcon.simlate('click');
  //   expect(onButtonClick).to.have.property('callCount', 1);
  // });
});
