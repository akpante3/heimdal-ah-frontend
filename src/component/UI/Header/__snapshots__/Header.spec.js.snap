// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App should match the snapshot 1`] = `
<header
  className="header"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <Link
      replace={false}
      to="/"
    >
      <img
        alt="heimdal-logo"
        className="logo"
        src="/public/images/logo.png"
      />
    </Link>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <NavItems />
      </ul>
      <div
        className="header-button-group"
      >
        <Link
          replace={false}
          to="/signup"
        >
          <button
            className="header-button"
            type="button"
          >
            Signup
          </button>
        </Link>
        <Link
          replace={false}
          to="/login"
        >
          <button
            className="header-button"
            type="button"
          >
            Login
          </button>
        </Link>
      </div>
      <form
        className="header-form form-inline my-2 my-lg-0"
      >
        <div
          className="header-input"
        >
          <input
            aria-label="Search"
            className="form-control  mr-sm-2"
            placeholder="Search"
            type="search"
          />
        </div>
        <div
          className="search_icon"
          onClick={[Function]}
        >
          <i
            className="fas fa-search"
          />
        </div>
      </form>
    </div>
  </nav>
</header>
`;
